### PR TITLE
Update starknet-rs and devnet

### PR DESCRIPTION
**Pull request by THenry14**
This PR bumps starknet-rs and devnet to support rpc 0.5.1. This is needed for (among other issues) https://github.com/foundry-rs/starknet-foundry/issues/1327

## Introduced changes

- Bump starknet-rs and starknet-devnet deps

## Checklist

- [X] Linked relevant issue
- [X] Updated relevant documentation
- [X] Added relevant tests
- [X] Performed self-review of the code
- [X] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/sncast/tests/e2e/call.rs [hunk 1]
```diff
@@ -85,7 +85,7 @@ fn test_wrong_function_name() {
 
     snapbox.assert().stderr_matches(indoc! {r"
         command: call
-        error: Entry point [..] not found in contract.
+        error: Contract error
```
Comment (war-in) [inline crates/sncast/tests/e2e/call.rs:88]: I assume this reflects changes from the rpc update, right? Those old errors were nice. `Contract error` is not meaningful at all.

**Review by war-in — APPROVED** [review]
Approved, however previous errors were better 😢

File: crates/sncast/tests/e2e/call.rs (hunk 1 above)
Comment (THenry14) [reply crates/sncast/tests/e2e/call.rs:88]: that's true, I liked them better as well. I have created an issue (for now on our part) to talk to devnet guys and see if something's could be done with this.

Comment (DelevoXDG) [reply crates/sncast/tests/e2e/call.rs:88]: `Contract error` should also have `data` with `revert_error` that has more specific details, maybe it's just not currently handled?

**Pull request merged**